### PR TITLE
Restructtion and updates

### PR DESCRIPTION
**Pull request by Apollon77**
This PR can be best reviewed by commit.

It basically does:
* Splitup the one big file intop multiple to have a better structure
* Adds Prettier for conistent codestyle
* Removes publishing the legacy package, so we only go forward with the new one (@iobroker/type-detector)
* it fixes the Electricity state detection for light type
* it adds Electricity states to more types

### CODE REVIEW

**Comment by Apollon77** [comment]
I would release this as a major for easier upgrades.

BTW:
* new package use: [iobroker.matter](https://www.npmjs.com/package/iobroker.matter)
    [iobroker.hm-rpc](https://www.npmjs.com/package/iobroker.hm-rpc)
    [iobroker.scenes](https://www.npmjs.com/package/iobroker.scenes)
* old package use: [iobroker.lovelace](https://www.npmjs.com/package/iobroker.lovelace)
    [iobroker.iot](https://www.npmjs.com/package/iobroker.iot)

Additionally:
ioBroker.devices/src/package.json:        "iobroker.type-detector": "^1.1.2",
ioBroker.fullcalendar/src/package.json:        "iobroker.type-detector": "^3.0.5",
ioBroker.material/src/package.json:    "iobroker.type-detector": "^1.1.1",

plus admin most likely

File: package.json [hunk 1]
```diff
@@ -21,28 +21,30 @@
...
+    "mocha": "^10.7.0",
+    "prettier": "^3.3.3",
+    "prettier-plugin-organize-imports": "^4.0.0",
+    "typescript": "~5.5.4"
   },
   "bugs": {
     "url": "https://github.com/ioBroker/ioBroker.type-detector/issues"
   },
   "main": "index.js",
   "files": [
```
Comment (foxriver76) [inline package.json:43]: Puh for me it is a bit strange, that this `build.js` script copies the package.json into `build` and publish then happens from there. During cleanup I would try to achieve publish from root dir then just publish `build` dir and `LICENSE`. What do you think?

**Review by foxriver76 — APPROVED** [review]

File: DEVICES.md [hunk 2]
```diff
@@ -74,21 +74,26 @@ In [brackets] is given the class name of device.
...
+|   | POWER          | switch.power                  |      | boolean/number | W  |     |      | ``/^switch\.power$/``                      |
+|   | POWER          |                               |      | boolean        | W  |     |      | ``/^switch$/``                             |
+|   | ACTUAL         | value.temperature             | °C   | number         | -  |     |      | ``/temperature(\..*)?$/``                  |
+|   | HUMIDITY       | value.humidity                | %    | number         | -  |     |      | ``/humidity(\..*)?$/``                     |
+|   | BOOST          | switch.boost                  |      | boolean/number | W  |     |      | ``/^switch\.boost(\..*)?$/``               |
+|   | SWING          | level.mode.swing              |      | number         | W  |     |      | ``/swing$/``                               |
+|   | SWING          | switch.mode.swing             |      | boolean        | W  |     |      | ``/swing$/``                               |
+|   | ELECTRIC_POWER | value.power                   | W    | number         | -  |     |      | ``/^value\.power$/``                       |
```
Comment (Garfonso) [inline DEVICES.md:89]: Maybe we should adjust the script that produces the devices.md to separate those shared patterns? So they do not show up on every device. (Even more for unreach and so on)... 
Or is, there, perhaps, an easy way to hide the lines by default?

File: package.json (hunk 1 above)
Comment (Apollon77) [reply package.json:43]: As discussed offline we keep that way for now because it removed devDeps

File: DEVICES.md (hunk 2 above)
Comment (Apollon77) [reply DEVICES.md:89]: Can you create an issue? But in general it must not all shared be present .. sooo good question whats best

**Pull request merged**